src/lib.rs

```diff
@@ -115,23 +115,15 @@ pub mod hash_types;
 pub mod policy;
 
 pub use crate::hash_types::*;
-pub use crate::blockdata::block::Block;
-pub use crate::blockdata::block::BlockHeader;
+pub use crate::blockdata::block::{Block, BlockHeader};
 pub use crate::blockdata::script::Script;
-pub use crate::blockdata::transaction::Transaction;
-pub use crate::blockdata::transaction::TxIn;
-pub use crate::blockdata::transaction::TxOut;
-pub use crate::blockdata::transaction::OutPoint;
-pub use crate::blockdata::transaction::EcdsaSighashType;
+pub use crate::blockdata::transaction::{EcdsaSighashType, OutPoint, TxIn, TxOut, Transaction};
 pub use crate::blockdata::witness::Witness;
 pub use crate::consensus::encode::VarInt;
 pub use crate::network::constants::Network;
 pub use crate::util::Error;
-pub use crate::util::address::Address;
-pub use crate::util::address::AddressType;
-pub use crate::util::amount::Amount;
-pub use crate::util::amount::Denomination;
-pub use crate::util::amount::SignedAmount;
+pub use crate::util::address::{Address, AddressType};
+pub use crate::util::amount::{Amount, Denomination, SignedAmount};
 pub use crate::util::merkleblock::MerkleBlock;
 pub use crate::util::sighash::SchnorrSighashType;
 
```

Comment:
nit: Can remove this newline

Happy with group by module!

Reply:
Thanks for the review man, and for checking my changes carefully. I'll just leave it as is for now, if you don't mind, for the reason layed out below (the `io` comment). Cheers

Reply:
You're welcome! 

Got it! `rustfmt` coming in (eventually) will be great so we can mostly forget these things :D